ci_watson/artifactory_helpers.py

```diff
@@ -124,6 +152,309 @@ def get_bigdata(*args):
+                             self.ignore_keywords+['NAXIS1', 'TFORM*'],
+                             'ignore_fields':self.ignore_keywords}
+                   }
+                  ]
+    .. NOTE::
+    Note that each entry in the list gets interpreted and processed
+    separately.
+    """
+    all_okay = True
+    creature_report = ''
+    # Create instructions for uploading results to artifactory for use
+    # as new comparison/truth files
+    testpath, testname = os.path.split(os.path.abspath(os.curdir))
+    # organize results by day test was run...could replace with git-hash
+    whoami = getpass.getuser() or 'nobody'
+    dt = datetime.datetime.now().strftime("%d%b%YT")
+    ttime = datetime.datetime.now().strftime("%H_%M_%S")
+    user_tag = 'NOT_CI_{}_{}'.format(whoami, ttime)
+    build_tag = os.environ.get('BUILD_TAG',  user_tag)
+    build_suffix = os.environ.get('BUILD_MATRIX_SUFFIX', 'standalone')
+    testdir = "{}_{}_{}".format(testname, build_tag, build_suffix)
+
+    # Parse any user-specified kwargs
+    ignore_keywords = kwargs.get('ignore_keywords', [])
+    ignore_hdus = kwargs.get('ignore_hdus', [])
+    ignore_fields = kwargs.get('ignore_fields', [])
+    rtol = kwargs.get('rtol', None)
+    atol = kwargs.get('atol', None)
+    input_path = kwargs.get('input_path', [])
+    results_root = kwargs.get('results_root', None)
+    docopy = kwargs.get('docopy', True)
+
+    updated_outputs = []
+    extn_list = None
+    for entry in outputs:
+        num_entries = len(entry)
+        if isinstance(entry, dict):
+            actual = entry['files'][0]
+            desired = entry['files'][1]
+            diff_pars = entry['pars']
+            ignore_keywords = diff_pars.get('ignore_keywords', ignore_keywords)
+            ignore_hdus = diff_pars.get('ignore_hdus', ignore_hdus)
+            ignore_fields = diff_pars.get('ignore_fields', ignore_fields)
+            rtol = diff_pars.get('rtol', rtol)
+            atol = diff_pars.get('atol', atol)
+        elif num_entries == 2:
+            actual, desired = entry
+        elif num_entries == 3:
+            actual, desired, extn_list = entry
+
+        if actual.endswith(']'):
+            actual_name, actual_extn = actual.split('[')
+            actual_extn = actual_extn.replace(']','')
+        else:
+            actual_name = actual
+            actual_extn = None
+
+        if desired.endswith(']'):
+            desired_name, desired_extn = desired.split('[')
+            desired_extn = desired_extn.replace(']','')
+        else:
+            desired_name = desired
+            desired_extn = None
+
+        # Get "truth" image
+        s = get_bigdata(*input_path, desired_name,
+                        docopy=docopy)
+        if s is not None:
+            desired = s
+            if desired_extn is not None:
+                desired = "{}[{}]".format(desired, desired_extn)
+        print("\nComparing:\n {} \nto\n {}".format(actual, desired))
```

Comment:
Maybe also something for the future to use logger instead of printing (or at least a non-verbose option to not print). Actually I am not even sure how useful printing this is because when FITSDiff or ASCII diff fails, it tells you what both files are anyway.